tidy3d/components/medium.py

```diff
@@ -1101,13 +1719,124 @@ def pole_residue(self):
+    >>> eps_inf = SpatialDataArray(np.ones((5, 6, 7)), coords=coords)
+    >>> d_epsilon = SpatialDataArray(np.random.random((5, 6, 7)), coords=coords)
+    >>> f = SpatialDataArray(1+np.random.random((5, 6, 7)), coords=coords)
+    >>> delta = SpatialDataArray(np.random.random((5, 6, 7)), coords=coords)
+    >>> lorentz_medium = CustomLorentz(eps_inf=eps_inf, coeffs=[(d_epsilon,f,delta),])
+    >>> eps = lorentz_medium.eps_model(200e12)
+    """
+
+    eps_inf: SpatialDataArray = pd.Field(
+        ...,
+        title="Epsilon at Infinity",
+        description="Relative permittivity at infinite frequency (:math:`\\epsilon_\\infty`).",
+        units=PERMITTIVITY,
+    )
+
+    coeffs: Tuple[Tuple[SpatialDataArray, SpatialDataArray, SpatialDataArray], ...] = pd.Field(
+        ...,
+        title="Coefficients",
+        description="List of (:math:`\\Delta\\epsilon_i, f_i, \\delta_i`) values for model.",
+        units=(PERMITTIVITY, HERTZ, HERTZ),
+    )
+
+    @pd.validator("eps_inf", always=True)
+    def _eps_inf_positive(cls, val):
+        """eps_inf should be positive"""
+        if not CustomDispersiveMedium._validate_isreal_dataarray(val):
+            raise SetupError("'eps_inf' should be real.")
+        if np.any(val < 0):
+            raise SetupError("'eps_inf' should be positive.")
+        return val
+
+    @pd.validator("coeffs", always=True)
+    def _coeffs_unequal_f_delta(cls, val):
+        """f and delta cannot be exactly the same.
+        Not needed for now because we have a more strict
+        validator `_coeffs_delta_all_smaller_or_larger_than_fi`.
+        """
+        return val
+
+    @pd.validator("coeffs", always=True)
+    def _coeffs_correct_shape_and_sign(cls, val, values):
+        """coeffs should have consistent shape and sign."""
+        if values.get("eps_inf") is None:
+            raise ValidationError("'eps_inf' failed validation.")
+
+        expected_coords = values["eps_inf"].coords
+        for (de, f, delta) in val:
+            if (
+                de.coords != expected_coords
+                or f.coords != expected_coords
+                or delta.coords != expected_coords
+            ):
+                raise SetupError(
+                    "All terms in 'coeffs' should have the same coordinates; "
+                    "The coordinates should also be consistent with 'eps_inf'."
+                )
+            if not CustomDispersiveMedium._validate_isreal_dataarray_tuple((de, f, delta)):
+                raise SetupError("All terms in 'coeffs' should be real.")
+        return val
+
+    @pd.validator("coeffs", always=True)
+    def _coeffs_delta_all_smaller_or_larger_than_fi(cls, val):
+        """We restrict either all f>delta or all f<delta for now."""
+        for (_, f, delta) in val:
+            if not (Lorentz._all_larger(f, delta) or Lorentz._all_larger(delta, f)):
+                raise SetupError("We restrict either all 'delta<f' or all 'delta>f'.")
```

Comment:
Very minor comments about error messages:
 - We don't tend to use "we" in error messages. How about "All coefficients must obey either 'delta' < 'f' everywhere, or 'delta' > 'f' everywhere." Note also the slightly different formatting of the inequality statement. I think we use `'` around variables, not around code-like statements like we would in markdown.
 - Could you change "should" to "must" in all messages?

Reply:
Thanks, both have been addressed in the latest commit.